### PR TITLE
Add Missing Diagnostic Settings to Modules

### PR DESCRIPTION
**Pull request by senavar**
# Change

>Thank you for your contribution !
Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Updated the Log Analytics, SQL DB, and Storage Account modules to allow for the configuration of diagnostic settings 

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

[OperationalInsights: Workspaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.operationalinsights.workspaces.yml)

[Sql: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.servers.yml)

[Storage: StorageAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.storage.storageaccounts.yml)

### CODE REVIEW

**Comment by MrMCake** [comment]
Hey @senavar, any chance you tested the code in pipelines? If so, could you do me a favor and add the pipeline badges to your PR summary for the pipelines to which you added the diagnostic settings?

**Review by MrMCake — COMMENTED** [review]
Hey @senavar, any chance you tested the code in pipelines? If so, could you do me a favor and add the pipeline badges to your PR summary for the pipelines to which you added the diagnostic settings?

File: arm/Microsoft.OperationalInsights/workspaces/deploy.bicep [hunk 1]
```diff
@@ -54,6 +54,23 @@ param publicNetworkAccessForQuery string = 'Enabled'
 @description('Optional. Set to \'true\' to use resource or workspace permissions and \'false\' (or leave empty) to require workspace permissions.')
```
Comment (MrMCake) [inline arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:54]: For all: @senavar, any chance you tested the code in pipelines? If so, could you do me a favor and add the pipeline badges to your PR summary for the pipelines to which you added the diagnostic settings?

Comment (senavar) [reply arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:54]: Badges have been added!

**Review by MrMCake — APPROVED** [review]

**Pull request merged**